cmd/kube-controller-manager/app/controllermanager_test.go

```diff
@@ -115,6 +121,56 @@ func TestNewControllerDescriptorsShouldNotPanic(t *testing.T) {
 	NewControllerDescriptors()
 }
 
+func TestGracefulControllers(t *testing.T) {
+	controllerDescriptorMap := NewControllerDescriptors()
+	gracefulControllers := sets.NewString(names.ClusterRoleAggregationController)
+
+	for controllerName, controllerDesc := range controllerDescriptorMap {
+		t.Run(controllerName, func(t *testing.T) {
+			defer goleak.VerifyNone(t)
```

Comment:
We could also just use `testing/synctest`, but I guess this is being used in Kubernetes elsewhere as well.

Reply:
`VerifyNone` runs as soon as it's invoked and doesn't wait for other goroutines to terminate. Obviously, if races happen then we could still have a passing result, but at least it will fail a percentage of the time.

Reply:
For instance, `VerifyNone` caught a bug with the typed queue not terminating properly (`ShutDown()` can complete before the goroutine is closed). This wasn't seen with `synctest`